rfc/rfc-37/rfc-37.md

```diff
@@ -0,0 +1,264 @@
+cardinality and the collision probability desired for the support max scale deployment. Base64 enoding the hash IDs
+further reduces the on-disk storage space for these keys.
+
+```
+key = base64_encode(hash64(partition name) + hash128(file path))
+```
+
+<img src="metadata_index_bloom_partition.png" alt="Bloom filter partition" width="500"/>
+
+### MetaIndex/ColumnStats:
+
+Another new partition `column_stats` will also be added under the metadata table to make the record key lookup code path
+much more performant. This metadata index also helps in the data skipping (please look at RFC-27 for more details). In
+the context of faster record key lookups for the update use cases, proposing `column_stats` index to be used for
+file pruning when generating the file to candidate keys mapping for the update records.The existing metadata payload
+schema will be extended and shared for this partition also. The type field will be used to detect the column stats
+payload record. Here is the schema for the column stats payload record.
+
+```
+        {
+            "name": "ColumnStatsMetadata",
+            "doc": "Contains information about column ranges for all data files in the table",
+            "type": [
+                "null",
+                {
+                    "type": "record",
+                    "name": "HoodieColumnStats",
+                    "fields": [
+                        {
+                            "name": "rangeLow",
+                            "type": [
+                                "null",
+                                "string"
+                            ],
+                            "doc": "Low end of the range. For now, this is a String. Based on main data table schema, we can convert it to appropriate type"
+                        },
+                        {
+                            "name": "rangeHigh",
+                            "type": [
+                                "null",
+                                "string"
+                            ],
+                            "doc": "High end of the range. For now, this is a String. Based on main data table schema, we can convert it to appropriate type"
+                        },
+                        {
+                            "name": "isDeleted",
+                            "type": "boolean",
+                            "doc": "True if this file has been deleted"
+                        }
+                    ]
+                }
+            ]
+        }
+```
+
+Column stats records hold key ranges (min and max) for the file. The key for the column stat record would be an
+encoded string representing the tuple set of column name, partition name and the base file. The string names of
+these fields are converted to deterministic hash based IDs, and then they are base64 encoded, just like the
+bloom filter key.
+
+```
+key = base64_encode(hash64(column name) + hash64(partition name) + hash128(file path))
```

Comment:
Can you add some details on how column renames, etc will pan out in future wrt hash based keys here, when such schema evolutions are added to hudi.

Reply:
Added a section for the schema evolution.